base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/editing/iocs/dialog/AddIocDialog.java

```diff
@@ -0,0 +1,187 @@
+        @Override
+        public void widgetSelected(SelectionEvent e) {
+            nextPage();
+        }
+    };
+
+    private SelectionListener prevListener = new SelectionAdapter() {
+        @Override
+        public void widgetSelected(SelectionEvent e) {
+            previousPage();
+        }
+    };
+
+    /**
+     * Sets dialog content to the first page.
+     */
+    private void previousPage() {
+        updateStack(addIocPanel);
+        btnPrev.setVisible(false);
+        btnOk.addSelectionListener(nextListener);
+        btnOk.setData(IDialogConstants.NO_ID);
+        btnOk.setText("Next");
+    }
+
+    /**
+     * Sets dialog content to the second page.
+     */
+    private void nextPage() {
+        tempIoc = addViewModel.getSelectedIoc();
+        editIocPanel.setIOC(tempIoc);
+        updateStack(editIocPanel);
+        btnPrev.setVisible(true);
+        btnOk.removeSelectionListener(nextListener);
+        btnOk.setData(IDialogConstants.OK_ID);
+        btnOk.setText(IDialogConstants.OK_LABEL);
+    }
+
+    /**
+     * Method for swapping content in the dialog window when moving between
+     * pages.
+     * 
+     * @param top
+     *            The panel to be displayed.
+     */
+    private void updateStack(final Control top) {
+        DISPLAY.asyncExec(new Runnable() {
+            @Override
+            public void run() {
+                stack.topControl = top;
+                content.layout();
+            }
+        });
+    }
+
+    @Override
+    protected void createButtonsForButtonBar(Composite parent) {
+        btnPrev = createButton(parent, IDialogConstants.NO_ID, "Previous", false);
+        btnPrev.addSelectionListener(prevListener);
+        btnPrev.setVisible(false);
+
+        btnOk = createButton(parent, IDialogConstants.NO_ID, "Next", false);
+        btnOk.addSelectionListener(nextListener);
+        btnOk.setEnabled(false);
+
+        createButton(parent, IDialogConstants.CANCEL_ID, IDialogConstants.CANCEL_LABEL, false);
+
+        // Disable OK button if no IOC is selected.
+        addViewModel.addPropertyChangeListener("selectedName", new PropertyChangeListener() {
+
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                String name = (String) evt.getNewValue();
+                if (!Strings.isNullOrEmpty(name)) {
+                    btnOk.setEnabled(true);
+                    btnOk.setFocus();
+                } else {
+                    btnOk.setEnabled(false);
+                }
+            }
+        });
+
+        // Enables selection by double click
+        addViewModel.addPropertyChangeListener("confirmed", new PropertyChangeListener() {
+
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                nextPage();
+            }
+        });
+    }
+
+    @Override
+    protected Control createDialogArea(Composite parent) {
```

Comment:
This seems to share quite a bit of code with the same method in the parent, can we reduce the duplication?